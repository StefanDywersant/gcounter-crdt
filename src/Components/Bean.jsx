import React from 'react';
import PropTypes from 'prop-types';
import styles from './Bean.scss';

/**
 * Renders single bean
 */
const Bean = ({slideX, slideY}) => (
	<div
		className={styles.bean}
		style={{
			marginTop: Math.floor(slideY * 5) + 'px',
			marginLeft: Math.floor(slideX * 5) + 'px'
		}}
	/>
);


Bean.defaultProps = {
	slideX: 0,
	slideY: 0
};


Bean.propTypes = {

	/**
	 * Horizontal slide of bean (0-1)
	 */
	slideX: PropTypes.number,

	/**
	 * Vertical slide of bean (0-1)
	 */
	slideY: PropTypes.number

};


export default Bean;