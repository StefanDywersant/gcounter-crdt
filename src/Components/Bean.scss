.bean {
	width: 5px;
	height: 5px;
	transition: margin-top .5s, margin-left .5s;

	background: #0f0;
	border: 1px solid #009100;
	border-radius: 50%;
}