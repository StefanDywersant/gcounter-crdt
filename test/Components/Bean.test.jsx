import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Bean from '../../src/Components/Bean';

const renderer = new ReactShallowRenderer();

describe('Component rendering', () => {

	it('renders correctly with default props', () => {
		expect(renderer.render(
			<Bean />
		)).toMatchSnapshot();
	});

	it('renders correctly with slides', () => {
		expect(renderer.render(
			<Bean slideX={.9} slideY={.3} />
		)).toMatchSnapshot();
	});

});
