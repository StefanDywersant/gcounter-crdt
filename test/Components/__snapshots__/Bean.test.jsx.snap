// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component rendering renders correctly with default props 1`] = `
<div
  className="bean"
  style={
    Object {
      "marginLeft": "0px",
      "marginTop": "0px",
    }
  }
/>
`;

exports[`Component rendering renders correctly with slides 1`] = `
<div
  className="bean"
  style={
    Object {
      "marginLeft": "4px",
      "marginTop": "1px",
    }
  }
/>
`;
